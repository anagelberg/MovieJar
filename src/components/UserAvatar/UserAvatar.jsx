import './UserAvatar.scss';

function UserAvatar({ img, ...attributes }) {
    return (
        <div className='avatar' {...attributes}>
            {img && <img src={img} alt="user avatar image" className='avatar__img' />}
        </div>
    )
}

export default UserAvatar;