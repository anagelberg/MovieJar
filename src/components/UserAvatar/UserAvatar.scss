@use "../../styles/variables" as *;

.avatar {
  height: $big-icon;
  width: $big-icon;
  border-radius: 50%;
  background-color: $primary;
  flex-shrink: 0;
  overflow: hidden;

  &__img {
    width: 100%;
    object-fit: cover;
    object-position: center;
  }
}
