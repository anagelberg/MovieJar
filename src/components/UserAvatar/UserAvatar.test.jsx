import UserAvatar from "./UserAvatar";
import { screen, render } from "@testing-library/react";

const testImg = "https://image.tmdb.org/t/p/original/3bhkrj58Vtu7enYsRolD1fZdja1.jpg"

test('renders UserAvatar component with img', () => {
    const { container } = render(<UserAvatar img={testImg} />);
    expect(screen.queryByRole('img')).toBeInTheDocument();
    expect(container.firstChild).toHaveClass('avatar');
});

test('renders UserAvatar component without img', () => {
    render(<UserAvatar />);
    expect(screen.queryByRole('img')).toBeNull();
});

test('UserAvatar component matches snapshot', () => {
    const { container } = render(<UserAvatar img={testImg} />);
    expect(container.firstChild).toMatchSnapshot();
});
//todo: add other without img tests