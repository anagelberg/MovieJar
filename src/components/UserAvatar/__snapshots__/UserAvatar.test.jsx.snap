// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar component matches snapshot 1`] = `
<div
  class="avatar"
>
  <img
    alt="user avatar image"
    class="avatar__img"
    src="https://image.tmdb.org/t/p/original/3bhkrj58Vtu7enYsRolD1fZdja1.jpg"
  />
</div>
`;
